address_book/src/signal.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Configurable Asset Privacy for Ethereum (CAPE) library.
+
+// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use async_std::task::spawn;
+use async_std::task::JoinHandle;
+use signal_hook::{consts::SIGINT, consts::SIGTERM, iterator::Signals};
+use std::process;
+
+/// Spawn a thread that waits for SIGTERM. If SIGTERM is received,
+/// the application exits with exit status 1.
+pub async fn init_sig_handler() -> Result<JoinHandle<()>, ()> {
+    let mut signals = Signals::new(&[SIGINT, SIGTERM]).unwrap();
+    Ok(spawn(async move {
```

Comment:
Because this is a blocking thread you should use [spawn_blocking](https://docs.rs/async-std/latest/async_std/task/fn.spawn_blocking.html)